### PR TITLE
Feature/include quote tweet

### PR DESCRIPTION
**Pull request by weswalla**

### CODE REVIEW

**Comment by weswalla** [comment]
@ronentk once this is live, you could handle it like so:

TYPESCRIPT SIDE:
```typescript
export interface GenericPostData {
  content: string;
  metadata: {
    originalPlatformAuthor: {
      id: string;
      username: string;
      name: string;
    };
    transcludedContent?: {
      url: string;
      content: string;
      author: {
        id: string;
        username: string;
        name: string;
      };
    }[];
  };
}
```

PYTHON SIDE:
```python
@https_fn.on_request(min_instances=1, memory=2048, timeout_sec=600)
def SM_FUNCTION_post_parser(request):
    """
    Wrapper on SM_FUNCTION_post_parser_imp
    """
    request_json = request.get_json()
    post = request_json["post"]
    parameters = request_json["parameters"]

    # we will handle credentials from here and also crendentials rotation if needed
    # input is going to be an array of posts to parse

    config: SM_FUNCTION_post_parser_config = {
        "openrouter_api_key": openai_api_key,
        "openrouter_api_base": "https://openrouter.ai/api/v1",
        "openrouter_referer": "https://127.0.0.1:3000/",
        "llm_type": model,
    }

    logger.info(f"Calling SM_FUNCTION_post_parser_imp with config: {config}")
    
    // ========= NEW CHANGES HERE ===========
    transcludedContent = post["metadata"]["transcludedContent"]
    // replace quote tweet urls with the content -- or however you want to include it
    newContent = replaceUrlWithQuoteTweet(post["content"],transcludedContent)

    parser_result = SM_FUNCTION_post_parser_imp(newContent, parameters, config)
    parser_json = parser_result.model_dump_json()

    return https_fn.Response(
        parser_json,
        status=200,
        headers={"Content-Type": "application/json"},
    )
```

File: app/webapp/src/shared/types/types.posts.ts [hunk 1]
```diff
@@ -7,6 +7,22 @@ import { PlatformPost } from './types.platform.posts';
  */
 export interface GenericPostData {
   content: string;
+  metadata: {
+    originalPlatformAuthor: {
```
Comment (pepoospina) [inline app/webapp/src/shared/types/types.posts.ts:11]: I would create a `GenericPostAuthor` interface which will be the author object we associate to any GenericPost, and will be the interface we use to talk about content and authors with the parser.

What about this interface?

```ts
export interface GenericPostAuthor {
  platformId: PLATFORM;
  id: string;
  username: string;
  name: string;
}

export interface GenericPostData {
  url?: string;
  author: GenericPostAuthor;
  content: string;
  quotedPosts?: GenericPostData[];
}

```

File: app/firebase/functions/test/__tests__/setup.ts [hunk 2]
```diff
@@ -58,12 +59,27 @@ export const mochaHooks = (): Mocha.RootHookObject => {
...
           const fileContents = fs.readFileSync(TEST_USERS_FILE_PATH, 'utf8');
           appUsers = JSON.parse(fileContents);
-
-          await Promise.all(
-            appUsers.map(async (appUser) => {
-              testUsers.set(appUser.userId, appUser);
-            })
-          );
+          if (
+            checkOutdatedTwitterTokens(appUsers) ||
```
Comment (pepoospina) [inline app/firebase/functions/test/__tests__/setup.ts:63]: In principle, even if expired, we should be able to use the refresh token (automatically) and update the access token as part of the test. At the end the test.users.json file should be rewritten with the new access token and refresh token.

For some reason this is not working, suggesting the refresh token stored in test.users.json is not the valud one.

**Review by pepoospina — COMMENTED** [review]
Hey, everything seems correct. I left some comments that I think are worth considering before moving on

File: app/webapp/src/shared/types/types.posts.ts [hunk 3]
```diff
@@ -7,6 +7,22 @@ import { PlatformPost } from './types.platform.posts';
...
+      id: string;
+      username: string;
+      name: string;
+    };
+    transcludedContent?: {
```
[the remark this answers is not in the document]
Comment (weswalla) [reply app/webapp/src/shared/types/types.posts.ts:16]: seems more clear

File: app/firebase/functions/src/platforms/twitter/twitter.utils.ts [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
   UserV2,
 } from 'twitter-api-v2';
 
+import {
+  AppTweet,
+  AppTweetBase,
+  OPTIONAL_TWEET_FIELDS,
+  REQUIRED_TWEET_FIELDS,
+  TwitterThread,
+} from '../../@shared/types/types.twitter';
```
Comment (weswalla) [inline app/firebase/functions/src/platforms/twitter/twitter.utils.ts:14]: @pepoospina this is where I was getting a strange error. If I kept the old path, I would get the error when running tests:
```
✖ ERROR: Error: Cannot find module 'src/@shared/types/types.twitter'
```

Which only came up if I was importing an `enum` (or a `const`) but worked fine for types and interfaces.

Comment (pepoospina) [reply app/firebase/functions/src/platforms/twitter/twitter.utils.ts:15]: It's a dumb error. The emulator forces us use relative paths, not absolute `/src`. You can replace `/src` with `../../ etc`

**Comment by pepoospina** [comment]
Have you tried running the tests (at least 030 and 050) @weswalla ? 
I'm seeing a weird error when running `yarn test`

```ts
✖ ERROR: ../../webapp/src/shared/types/types.twitter.ts:71:17 - error TS2344: Type 'OPTIONAL_TWEET_FIELDS' does not satisfy the constraint 'keyof TweetV2'.

71   Pick<TweetV2, OPTIONAL_TWEET_FIELDS>;
``

I merged into develop and pushed the partial merge status to
https://github.com/pepoospina/sensemakers/tree/include-quote-tweet-rev

Can you check it out?

**Comment by weswalla** [comment]
I'll take a look

**Comment by weswalla** [comment]
@pepoospina I don't get that error when I checkout your branch and run the tests. Those tests do fail though, but it doesn't seem like it has anything to do with this PR. I think its the mock threads which is stored in the `.env.test` since they mostly seem to be differences in expected length of returned arrays.

**Pull request merged**